presto-jdbc/src/main/java/com/facebook/presto/jdbc/QueryExecutor.java

```diff
@@ -42,7 +43,7 @@ public QueryExecutor(OkHttpClient httpClient)
 
     public StatementClient startQuery(ClientSession session, String query)
     {
-        return new StatementClient(httpClient, session, query);
+        return newStatementClient(httpClient, session, query);
     }
 
     public ServerInfo getServerInfo(URI server)
```

Comment:
I think you're suggesting wider changes to how clients are managed across CLI, jdbc, etc. It might be possible, but I'd rather do it as a different PR.